mapscript/swiginc/class.i

```diff
@@ -169,8 +169,17 @@
     }
     else
       layer->scalefactor = map->resolution/map->defresolution;
-    
+
+  #if defined(WIN32) && defined(SWIGCSHARP)
+    __try {
+        return msDrawLegendIcon(map, layer, self, width, height, dstImage, dstX, dstY, MS_TRUE, NULL);
```

Comment:
this looks like a workaround than a proper fix. Couldn't the exception be caught upwards?
But if you want to keep this, why not just using the _WIN32 canoncial symbol ? This would avoid defining WIN32 explicitly

Reply:
Similar workaround already exists in map.i that have been working for the client who has contributed that.
Using _WIN32 might also work, I haven't tested that. Also I didn't check who else is relying on WIN32 actually (it has been enabled in the nmake based builds for years).

Reply:
> Similar workaround already exists in map.i that have been working for the client who has contributed that.

hum, it would be good to understand what happens. It seems to me that the workaround, if really needed, should be moved to msGEOSOffsetCurve() itself, as I presume it affects the CGI mapserv too. Is this is a bug in GEOS (but the C functions of GEOS are supposed to catch C++ exceptions) ? or some issue with binaries not being compiled with consistent runtimes ?

Reply:
Looks like BufferBuilder::bufferLineSingleSided in geos throws util::IllegalArgumentException. I'm not sure who is responsible to catch this, but it brings down the entire IIS host process if that is not catched.

Reply:
GEOSOffsetCurve_r() (https://github.com/libgeos/geos/blob/7a10a9f1857f8ba0231f5d3a9022e5d959b89c50/capi/geos_ts_c.cpp#L1168) calls execute() which catches all C++ exceptions: https://github.com/libgeos/geos/blob/7a10a9f1857f8ba0231f5d3a9022e5d959b89c50/capi/geos_ts_c.cpp#L424
So C++ exceptions shouldn't reach mapserver

Reply:
This looks like the same in my geos version. It is strange why the exceptions are propagated in spite of that.

Reply:
It looks like a SEH (OS) exception is being generated with ex. code = 0XE06D7363 (I'm not quite sure why). But the default /EHsc option makes that it is not being catched by the C++ exception handlers.